### ISSUE TITLE
Remove use of qInfo() from linuxdeployqt

### ISSUE THREAD

**Issue by cgarry** [probonopd/linuxdeployqt#238]
qInfo() has only been included in Qt since version 5.5 and using it prevents linuxdeployqt from being build using earlier versions.

This is a particular issue for me as it means linuxdeployqt will not build on Ubuntu 14.04 without modifying the tools/linuxdeployqt/main.cpp to remove all the qInfo() calls.  Note that I only need to build linuxdeployqt this as there is no 32-bit AppImage for linuxdeployqt available.

**Comment by probonopd** [comment]
Please use the latest Qt that is available for trusty from the ppa as per the README.md. Earlier Qt versions have issues that prevent `linuxdeployqt` from working correctly, and I don't have the time to write workarounds for older Qt 5 versions; especially since the issues often have been resolved in Qt since then, and the fixes are available in the ppa for trusty.

**Comment by cgarry** [comment]
Thanks for the clarification.  I agree that it is perfectly reasonable to require a minimum version of Qt rather than putting workarounds in linuxdeployqt to support older versions.  The README.md file does not make it clear that a minimum version of Qt is required, at least it did not to me as I read it.

Is the minimum version of Qt just required for just the building of linuxdeploy or is it also required for the application being packaged?  What is the minimum version of Qt required?

For linuxdeployqt we should add something similar to the code below to the .pro file so that users cannot inadvertently build it with an incompatible version:

```text
contains(QT_VERSION, ^4\\.[0-6]\\..*) {
    message("Cannot build linuxdeployqt with Qt version $${QT_VERSION}.")
    error("Use at least Qt 4.7.")
}
```

For the application that is being packaged, I think that linuxdeployqt should check the version of Qt used for the application and stop with an error if the version is older than the minimum version.

If this makes sense I could look at making the relevant changes to linuxdeployqt.

**Comment by probonopd** [comment]
Thanks @cgarry, I'd appreciate if you could send a pull request. For Qt 5.x we should require 5.8 iirc.

**Comment by cgarry** [comment]
Are we sure about Qt 5.8 being the oldest version that can be used?  I ask because of this line in BUILDING.md:

> * Get and build linuxdeployqt e.g., using Qt 5.7.0 (you could use this [Qt Creator AppImage](https://bintray.com/probono/AppImages/QtCreator#files) for this)

**Comment by probonopd** [comment]
Right, at the time I wrote this I was using Qt 5.7.0; it may still be sufficient. But why not use the latest one available in the 5.x series as a ppa for trusty?

**Comment by TheAssassin** [comment]
@cgarry let us know if you tried and found a working minimum version.

**Comment by cgarry** [comment]
@probonopd 
> Right, at the time I wrote this I was using Qt 5.7.0; it may still be sufficient. But why not use the latest one available in the 5.x series as a ppa for trusty?

Indeed, for my AppImage I use the ppa (Qt 5.9) for trusty as you suggested and this is working perfectly.  I was asking as part of the task to add protection to the linuxdeployqt code so that it cannot be build with an unsuitable version of Qt.

@TheAssassin 
> let us know if you tried and found a working minimum version.

I was able to build both linuxdeployqt and my application with Qt 5.2.1 (the version that comes with trusty) and it seemed to work perfectly for me.  To do this however, I did have to comment out of the qinfo() instances in the linuxdeployqt code but it did function correctly as far as I could tell.

So I am at a bit of a loss as to what the issues are that @probonopd was referring to with this comment: "Earlier Qt versions have issues that prevent linuxdeployqt from working correctly" and which is the minimum version of Qt required.

**Comment by TheAssassin** [comment]
@cgarry is it _only_ `qInfo()` and not `qDebug` etc. which causes trouble? If yes, we should consider replacing this function. Could you count the occurrences, please?

**Comment by probonopd** [comment]
> So I am at a bit of a loss as to what the issues are that @probonopd was referring to with this comment: "Earlier Qt versions have issues that prevent linuxdeployqt from working correctly" and which is the minimum version of Qt required.

I can guarantee you I ran into issues with earlier versions, I think when I tried to bundle the QtWeb* stuff, and I noticed that the same issues do not occur with the latest versions of the Qt 5.x series. So I recommend to use that. Personally I don't have the time to investigate anything else but the latest Qt 5.x that is available for the oldest still-supported LTS, which at this time is trusty.
